internal/controller/promotions/promotions.go

```diff
@@ -84,7 +84,13 @@ func SetupReconcilerWithManager(
 		pqs:         reconciler.pqs,
 	}
 	promoWentTerminal := kargo.NewPromoWentTerminalPredicate(logger)
-	if err := c.Watch(&source.Kind{Type: &kargoapi.Promotion{}}, priorityQueueHandler, promoWentTerminal); err != nil {
+	if err := c.Watch(
+		source.Kind(kargoMgr.GetCache(),
+			&kargoapi.Promotion{},
+		),
```

Comment:
@jessesuen this is what I was talking about... the fact that you have to pick a cache (which is a wrapper around a client) when you call `Watch` makes me suspect we may be able to create controllers that reconcile using one client, but also watch things using another. (Useful for sharding.)